shark-log/src/main/java/shark/SharkLog.kt

```diff
@@ -13,45 +13,31 @@ object SharkLog {
     /**
      * Logs a debug message formatted with the passed in arguments.
      */
-    fun d(
-      message: String,
-      vararg args: Any?
-    )
+    fun d(message: String)
 
     /**
      * Logs a [Throwable] and debug message formatted with the passed in arguments.
      */
-    fun d(
-      throwable: Throwable,
-      message: String,
-      vararg args: Any?
-    )
+    fun e(throwable: Throwable, message: String)
   }
 
   @Volatile var logger: Logger? = null
 
   /**
    * @see Logger.d
    */
-  fun d(
-    message: String,
-    vararg args: Any?
-  ) {
+  inline fun d(message: () -> String) {
     // Local variable to prevent the ref from becoming null after the null check.
     val logger = logger ?: return
-    logger.d(message, *args)
+    logger.d(message.invoke())
   }
 
   /**
-   * @see Logger.d
+   * @see Logger.e
    */
-  fun d(
-    throwable: Throwable,
-    message: String,
-    vararg args: Any?
-  ) {
+  inline fun e(throwable: Throwable, message: () -> String) {
```

Comment:
I don't like how this method gets invoked, with two parameters like `SharkLog.e(exception) { "More text" }` 
Any suggestions on making it more friendly?

Reply:
Also naming. Switched from `d()` to `e()` to make it more explicit that this one is for errors and to prevent confusion when using autocomplete. Do we want to keep `e()`?

Reply:
Ah, I commented above on `e`.

I can't think of another API than the two params.